src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -47,7 +61,12 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
         const buttonParams: ButtonsOptions = {
             style: buttonStyle,
             onApprove: data => this._tokenizePayment(data, submitForm),
-            onClick: async (_, actions) => onValidate(actions.resolve, actions.reject),
+            onClick: async (_, actions) => {
+                this._initializePollingMechanism(submitForm, gatewayId);
```

Comment:
I assume polling is done because it's the only way to handle this situation? Also why do we still want to proceed with payment if the user explicitly cancels the transaction?

Reply:
yeah, this topic is partially covered in the [thread](https://threads.com/34389654791). If a buyer cancels a transaction, then the `onCancel` is triggered and we stop the initiated session (it is similar to the regular PayPal flow).

Reply:
I see, it's not "cancelling", but rather, it's about confirmation status being returned asynchronously due to the way APM works.